store/store.go

```diff
@@ -0,0 +1,142 @@
+package store
+
+import (
+	"encoding/binary"
+	"errors"
+	"sync"
+
+	"go.uber.org/multierr"
+
+	"github.com/lazyledger/optimint/hash"
+	"github.com/lazyledger/optimint/types"
+)
+
+var (
+	blockPrefix  = [1]byte{1}
+	indexPrefix  = [1]byte{2}
+	commitPrefix = [1]byte{3}
+)
+
+type DefaultStore struct {
+	db KVStore
+
+	height uint64
+
+	// mtx protects height
+	mtx sync.RWMutex
```

Comment:
nit: keeping the value and it's guardian close to each other increases readibility

Reply:
The comment is misleading. Mutex is used to ensure that `height` is in sync with `db`.